src/domain/staking/staking.repository.ts

```diff
@@ -93,6 +102,40 @@ export class StakingRepository implements IStakingRepository {
     return DefiVaultsStateSchema.parse(defiStats)[0];
   }
 
+  public async getDefiVaultStake(args: {
+    chainId: string;
+    safeAddress: `0x${string}`;
+    vault: `0x${string}`;
+  }): Promise<DefiVaultStake> {
+    const stakingApi = await this.stakingApiFactory.getApi(args.chainId);
+    const defiStakes = await stakingApi.getDefiVaultStakes(args);
+    // Cannot be >1 contract deployed at the same address so return first element
+    return (
+      DefiVaultStakesSchema.parse(defiStakes)
+        // Should not return non-matching chain_id but just in case
+        .filter((stake) => {
+          return stake.chain_id === Number(args.chainId);
+        })[0]
```

Comment:
nit: We could also check that `stake.owner === args.safeAddress`

Reply:
I checked the API and it only returns data for the chain/address so this is not necessary. I removed it entirely in 879497c177b4a19f1143158159b018d4cabcca38.